wolfcrypt/src/port/Espressif/esp_crt_bundle/esp_crt_bundle.c

```diff
@@ -0,0 +1,1380 @@
+#ifndef IS_WOLFSSL_CERT_BUNDLE_FORMAT
+    /* For reference only, the other cert bundles are structured differently!
+     * The others contain only a PARTIAL certificate, along with a name. */
+    #define BUNDLE_HEADER_OFFSET 2
+    #define CRT_HEADER_OFFSET 4
+#else
+    /* Note these are also set in [ESP-IDF]/components/esp-tls/esp_tls_wolfssl.c
+     * to avoid conflicts with other cert bundles that may, in theory,
+     * be enabled concurrently (NOT recommeneded).
+     *
+     * Ensure they exactly match here: */
+    #define BUNDLE_HEADER_OFFSET 2
+    #define CRT_HEADER_OFFSET 2
+#endif
+
+/* NOTE: Manually edit sort order in gen_crt_bundle.py
+ *
+ * The default is having the bundle pre-sorted in the python script
+ * to allow for rapid binary cert match search at runtime. The unsorted
+ * seach ALWAYS works, but when expecting a sorted search the python
+ * script MUST presort the data, oherwise the connection will likely fail.
+ *
+ * When debugging and using an unsorted bundle, define CERT_BUNDLE_UNSORTED
+ * Reminder: the actual sort occurs in gen_crt_bundly.py call from CMake. */
+/* #define CERT_BUNDLE_UNSORTED */
+
+
+/* Inline cert bundle functions performance hint unless otherwise specified. */
+#ifndef CB_INLINE
+    #define CB_INLINE inline
+#endif
+
+/* A "Certificate Bundle" is this array of [size] + [x509 CA List]
+ * certs that the client trusts: */
+extern const uint8_t x509_crt_imported_bundle_wolfssl_bin_start[]
+                     asm("_binary_x509_crt_bundle_wolfssl_start");
+
+extern const uint8_t x509_crt_imported_bundle_wolfssl_bin_end[]
+                     asm("_binary_x509_crt_bundle_wolfssl_end");
+
+/* This crt_bundle_t type must match other providers in esp-tls from ESP-IDF.
+ * TODO: Move to common header in ESP-IDF. (requires ESP-IDF modification).
+ * For now, it is here: */
+typedef struct crt_bundle_t {
+    const uint8_t **crts;
+    uint16_t num_certs;
+    size_t x509_crt_bundle_wolfssl_len;
+} crt_bundle_t;
+
+#ifdef CONFIG_WOLFSSL_CERTIFICATE_BUNDLE
+static esp_err_t wolfssl_esp_crt_bundle_init(const uint8_t *x509_bundle,
+                                             size_t bundle_size);
+static esp_err_t _esp_crt_bundle_is_valid = ESP_FAIL;
+#endif
+
+static crt_bundle_t s_crt_bundle;
+static esp_err_t _wolfssl_found_zero_serial = ESP_OK;
+
+static int _cert_bundled_loaded = 0;
+
+static int _crt_found = 0;
+
+static int _added_cert = 0;
+
+static int _need_bundle_cert = 0;
+
+
+/* Returns ESP_OK if there are no zero serial numbers in the bundle,
+ * OR there may be zeros, but */
+static CB_INLINE int wolfssl_found_zero_serial()
+{
+    return _wolfssl_found_zero_serial;
+}
+
+/* Returns:
+ *   1 if the cert has a zero serial number
+ *   0 if the cert has a non-zero serial number
+ * < 0 for error wolfssl\wolfcrypt\error-crypt.h values  */
+static CB_INLINE int wolfssl_is_zero_serial_number(const uint8_t *der_cert,
+                                                     int sz) {
```

Comment:
Brace on new line

Reply:
Missed that one; moved and aligned params